libraries/bot-azure/src/test/java/com/microsoft/bot/azure/TranscriptStoreTests.java

```diff
@@ -148,21 +147,27 @@ public void transcriptRemoveTest() {
 
     @Test
     public void activityAddSpecialCharsTest() {
+        TranscriptStore transcriptStore = getTranscriptStore();
         Activity[] loggedActivities = new Activity[CONVERSATION_SPECIAL_IDS.length];
         List<Activity> activities = new ArrayList<Activity>();
         for (int i = 0; i < CONVERSATION_SPECIAL_IDS.length; i++) {
             Activity a = TranscriptStoreTests.createActivity(i, i, CONVERSATION_SPECIAL_IDS);
             transcriptStore.logActivity(a).join();
             activities.add(a);
-            loggedActivities[i] = transcriptStore.getTranscriptActivities(channelId, CONVERSATION_SPECIAL_IDS[i])
-               .join().getItems().get(0);
+            int pos = i;
```

Comment:
Why this?

Reply:
I need to use the `i` variable inside the `thenApply` function (a lambda function), when I try, java complains that I need to create another variable and assign it `i`

Reply:
😖 , but ok